infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesClientFactory.java

```diff
@@ -129,11 +129,27 @@ protected OkHttpClient getHttpClient() {
     return httpClient;
   }
 
+  /**
+   * Unlike {@link #getHttpClient()} method, this method always returns an HTTP client that contains
+   * interceptors that augment the request with authentication information available in the global
+   * context.
+   *
+   * <p>Unlike {@link #getHttpClient()}, this method creates a new HTTP client instance each time it
```

Comment:
Is it already implemented or just planned?

Reply:
This is implemented only for OpenShift. Maybe this should throw `UnsupportedOperationException` or some such.. Good point.